### PR TITLE
TN-1418: Handle multiple unique emails gracefully

### PR DESCRIPTION
**Pull request by drryanjames**
The /unique_email API asserts when it finds more than one match for an email address. This throws a 500. Because the UI calling the API doesn't get a direct response, it allows the staff member to create a  patient with the duplicate email address.

These changes fail gracefully, returning False which allows the UI to pop up an error message that prevents a new user from being created.

This, however, is not a complete fix for TN-1418. It only prevents the UI from proceeding, but does not prevent a user from being created through the /account API. That work is fixed in this [PR](https://github.com/uwcirg/true_nth_usa_portal/pull/2629).

### CODE REVIEW

**Comment by drryanjames** [comment]
@pbugni, would you mind clarifying why you think my assessment isn't true? I agree that we should never allow the insertion of multiple emails and have separated that change into [this PR](https://github.com/uwcirg/true_nth_usa_portal/pull/2629). I also agree that the client should handle 500's gracefully. Including @achen2401 in the loop.

**Comment by achen2401** [comment]
@pbugni @drryanjames paul and ryan, when the API returns error (e.g. 500), the client side currently **does not** proceed with the unique email check - note the client side does not check the error code currently - maybe it should? - so it "does not know" why the API call failed.  A returned json of unique == True/False allows the client to reliably determine the uniqueness of an email and therefore prevent a staff member from creating a patient account with duplicate email if necessary - that is my thinking.

**Comment by pbugni** [comment]
@drryanjames @achen2401 two issues.
1. I think it'd be reasonable for an API call to verify a valid response, but I'll support this change if it makes it easier.
2. I do NOT think this is the source of the problem, as the assertion will only fail *after* the actual insertion.  It is my understanding the unique_email API would get called *before* attempting to call /demographics and add it, no?

**Comment by achen2401** [comment]
paul, think you are right regarding #2. I will defer to you guys on what the best fix is here.  think the missing piece here is for the client side to display error message if the unique email API check fail and log the error. I think I will create a PR for that.

**Comment by drryanjames** [comment]
@pbugni, this is not the source of the error. The source of the error is fixed in my other PR.

File: tests/test_user.py [hunk 1]
```diff
@@ -107,6 +107,23 @@ def test_unique_email(self):
...
+        # we should fail gracefully
+        email = 'example@gmail.com'
+        firstMatch = self.add_user(username='foo', email=email)
+        secondMatch = self.add_user(username='bar', email=email.upper())
+        request = '/api/unique_email?email={}'.format(urllib.parse.quote(
+            email))
+        response = self.client.get(request)
+        assert response.status_code == 200
+        results = response.json
+        results['unique'] is False
```
[the remark this answers is not in the document]
Comment (drryanjames) [reply tests/test_user.py]: yes :/. Copy paste error

File: tests/test_user.py [hunk 2]
```diff
@@ -107,6 +107,23 @@ def test_unique_email(self):
...
+        # If there are more than two emails that match
+        # we should see false
+        # This scenario should not happen, but when it does
+        # we should fail gracefully
+        email = 'example@gmail.com'
+        firstMatch = self.add_user(username='foo', email=email)
+        secondMatch = self.add_user(username='bar', email=email.upper())
+        request = '/api/unique_email?email={}'.format(urllib.parse.quote(
+            email))
+        response = self.client.get(request)
```
[the remark this answers is not in the document]
Comment (drryanjames) [reply tests/test_user.py]: absolutely

**Pull request merged**